### PR TITLE
Add SearchOption for Km

### PR DESCRIPTION
**Pull request by Phil9219**

### CODE REVIEW

File: client/src/components/Page/ChooseYourRunPage.js [hunk 1]
```diff
@@ -28,10 +29,15 @@ export default function ChooseYourRunPage() {
...
 
+  function handleTypeIn(searchResults) {
+    setRuns(searchResults);
+  }
+
   return (
     <>
       <AppHeader title={"Choose Your Run"} />
       <ChooseRunContainer>
+        <Search handleSubmit={handleTypeIn} />
```
Comment (marwinburesch) [inline client/src/components/Page/ChooseYourRunPage.js]: `handleSubmit`, in this case would usually be called `onSubmit` if it is a function, that is called when something is submitted. That way, when implemented, the prop will look like: `onSubmit={handleSubmit}`, or in your case maybe `onSubmit={ (searchResults) => setRuns(searchResults) }` because that is what you want to do on submit :)

File: client/src/components/Search/Search.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+    <Form
+      onSubmit={async (event) => {
+        event.preventDefault();
+        if (searchKm) {
+          handleSubmit(await searchForKm(searchKm));
+        } else {
+          handleSubmit(await getRuns());
+        }
+      }}
```
Comment (marwinburesch) [inline client/src/components/Search/Search.js]: I wouldn't expect to have the logic for this here, at least looking at the implementation of the component so far. However, after closer examination, this Form doesn't look like it would need to have its logic externally, because you want to handle the data and then do something with it, outside this component.
**TL;DR:** 😂 Anyway, consider moving this to a function for readability

File: client/src/components/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        step="any"
+        autoComplete="off"
+        onChange={async (event) => {
+          setSearchKm(event.target.value);
+          if (event.target.value) {
+            handleSubmit(await searchForKm(event.target.value));
+          } else {
+            handleSubmit(await getRuns());
+          }
+        }}
```
Comment (marwinburesch) [inline client/src/components/Search/Search.js]: same here

**Review by marwinburesch — CHANGES_REQUESTED** [review]
Noice mate! 🐜

File: client/src/components/Page/ChooseYourRunPage.js (hunk 1 above)
Comment (Phil9219) [reply client/src/components/Page/ChooseYourRunPage.js]: Thanks
